sklearn/utils/extmath.py

```diff
@@ -325,13 +410,22 @@ def randomized_svd(M, n_components, n_oversamples=10, n_iter='auto',
         n_iter = 7 if n_components < .1 * min(M.shape) else 4
 
     if transpose == 'auto':
-        transpose = n_samples < n_features
+        if preconditioner == 'lobpcg':
+            transpose = n_samples > n_features
+        else:
+            transpose = n_samples < n_features
     if transpose:
-        # this implementation is a bit faster with smaller shape[1]
+        # this implementation is a bit faster with smaller shape [1]
         M = M.T
 
-    Q = randomized_range_finder(M, n_random, n_iter,
-                                power_iteration_normalizer, random_state)
+    if preconditioner is None:
```

Comment:
@glemaitre I would rather test the opposite: `if preconditioner == "lobpcg": ...` and else no preconditioner.